data/remnant/remnant 2 cognizance.txt

```diff
@@ -867,6 +867,25 @@ mission "Remnant: Cognizance 21"
 			`	The <ship> settles into the current landing pattern for the main spaceport on <destination> as you flip through the exterior camera feeds. In the distance, you notice that one of the research hangars has some activity going on around it, but otherwise, the spaceport seems to be humming along as usual. Dusk stops by the cockpit as you settle the <ship> into its berth. "Thank you for helping us, <first>. It will take us some time to analyze this data, but we will likely be mounting another expedition to Ssil Vida in the near future." He pauses, then adds, "I do not think we ever mentioned it, but, as is appropriate for your assistance, a logistics adjustment of <payment> has been applied to your account to cover your costs and provide resources to prepare for future endeavors." Dusk gives you a polite nod before heading off to help the other researchers unload the equipment.`
 
 
+mission "Remnant: Cognizance Exploration Patch"
+	landing
+	invisible
+	to offer
+		has explorationinterest
+	on offer
+		"exploration interest" = explorationinterest
+		clear explorationinterest
+		fail
+
+mission "Remnant: Cognizance Science Patch"
+	landing
+	invisible
+	to offer
+		has scienceinterest
+	on offer
+		"science interest" = scienceinterest
+		clear scienceinterest
+		fail
```

Comment:
These can just be one mission:
```suggestion

mission "Remnant: Cognizance Interest Patch"
	landing
	invisible
	to offer
		or
			has "explorationinterest"
			has "scienceinterest"
	on offer
		"exploration interest" += explorationinterest
		clear "explorationinterest"
		"science interest" += scienceinterest
		clear "scienceinterest"
		fail


```
Could even do:
`"explorationinterest" + "scienceinterest" > 1`
in the `to offer` condition if you want.

Reply:
What if you're in the middle of a mission that gives you, say `scienceinterest` but already have `explorationinterest` when this patch comes out?

Reply:
That's a good point.

Reply:
> What if you're in the middle of a mission that gives you, say `scienceinterest` but already have `explorationinterest` when this patch comes out?

I don't really get it.

Reply:
If you have an active mission that sets one of the old variables at the end of it, and already have the other, you could trigger the combined mission. Then, you would get the one you already have transferred over properly, but the other (old) one would still be set at the end of the mission, and then would never be changed, because active missions get stored in your save rather than referring back to the data files.

Reply:
How could that be fixed?

Reply:
You could set the fix mission to twist so that it triggers whenever someone has the old variable. Also supports plugins using the old variable too, that way .